MapboxNavigation/NavigationViewController.swift

```diff
@@ -46,7 +43,18 @@ public protocol NavigationViewControllerDelegate: VisualInstructionDelegate {
      */
     @objc(navigationViewController:didArriveAtWaypoint:)
     optional func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt waypoint: Waypoint) -> Bool
-
+    
+    /**
+     Called when the user arrives at the final destination of the current route leg and may display a feedback
+     UI to the user.
+     
+     If implemented, you can use this to detect the status of `CarPlayManager.isConnected` when the user arrives at the final destination.
+   
+     - parameter navigationViewController: The navigation view controller that has arrived at the leg of route.
+     - returns: This will most likely be used to determine whether to show feedback when the user arrives at the final leg of the route.
+     */
+    @objc optional func navigationViewControllerShouldShowFeedback(_ navigationViewController: NavigationViewController) -> Bool
```

Comment:
This seems redundant to `NavigationViewController.showsEndOfRouteFeedback`.

Reply:
`NavigationViewController.showsEndOfRouteFeedback` was replaced by this delegate.
Previously, `showsReportFeedback` was coupled with `showsEndOfRouteFeedback` so you couldn't pick one _or_ the other.

<details>

```swift
@objc public var showsReportFeedback: Bool = true {
        didSet {
            mapViewController?.reportButton.isHidden = !showsReportFeedback
            showsEndOfRouteFeedback = showsReportFeedback
        }
    }
```

</details>

`showsReportFeedback` simply toggles the visibility of the "realtime" feedback button.
Since CarPlay can be connected and disconnected throughout the session, we need to check that with the delegate.

However, this is a good indicator that the naming here isn't clear enough.
`s/navigationViewControllerShouldShowFeedback/navigationViewControllerShouldShowEndOfRouteFeedback`